### PR TITLE
Completed Serial.Write() Table

### PR DESCRIPTION
**Pull request by Boomstick105**
Creats a table for Serial.Write() commands

Notes;
For the Component.cpp I did this for now, if we need to figure out what out in the comment section, I'll work on this asap, but for now I got most of the working table up and running.

Checked it a couple times, hopefully this is in the correct format and typo free, if not I'll look into addressing what needs to be addressed.
 
Serial.write(13250); // 13250 = Log, Debug, Component object successfully created
//Need to figure out if we can put the component type created and the pin it's
  //set at with the Serial.write and the table, for now just doing base message
  //Logger::Log("Component " + componentType + " is set at pin " + pin);

### CODE REVIEW

File: src/Alarm.cpp [hunk 1]
```diff
@@ -13,24 +21,36 @@
...
+*/
 void Alarm::calibrate(){
   determineBasePhotoresistorReading();
   if(_photoR->takeReading() - _threshold < _baseReading){
     alertFailedAction();
     _isCalibrated = false;
   } else {
     alertSuccessfulAction();
-    Logger::Log("Alarm Successful Calibration");
+    Serial.write(30004); // 30004 = Command, N/A, Execute Command: Calibrate
```
Comment (Lastrellik) [inline src/Alarm.cpp]: The 30004 command is the command I send from the PI to the Arduino when I want it to calibrate. This isn't the right 5 digit number.

File: src/Alarm.cpp [hunk 2]
```diff
@@ -53,91 +80,138 @@ void Alarm::determineBasePhotoresistorReading(){
...
 }
-
+/**
+  Sets the alarm to be triggered
+  @param sets _isTriggered to be true
+  @param sets _isSilenced to be false
+*/
 void Alarm::trigger(){
-  Logger::Log("Alarm has been triggered!");
+  Serial.write(30005); // 30005 = Command, N/A, Execute Command: Trigger
```
Comment (Lastrellik) [inline src/Alarm.cpp]: Again, 000 - 099 are for commands that the PI sends back to the arduino to tell it to do something.

File: src/Alarm.cpp [hunk 3]
```diff
@@ -53,91 +80,138 @@ void Alarm::determineBasePhotoresistorReading(){
...
   if (not this->_isSilenced) _buzzer->soundAlarmLowTone();
   _alarmLED->flash(300);
 }
-
+/**
+  Disarms the alarm and sets the calibration to false to resetCalibration
+  @param _isCalibrated is set to false
+*/
 void Alarm::resetCalibration(){
+  Serial.write(30006); // 30006 = Commmand, N/A, Execute Commmand: Reset Calibration
```
Comment (Lastrellik) [inline src/Alarm.cpp]: Same as the above issues.

File: src/Alarm.cpp [hunk 4]
```diff
@@ -53,91 +80,138 @@ void Alarm::determineBasePhotoresistorReading(){
...
+*/
 void Alarm::disarm(){
   this->alertSuccessfulAction();
   _isTriggered = false;
   _isArmed = false;
   _alarmLED->off();
   _isSilenced = false;
   _laser->off();
-  Logger::Log("Alarm has been disarmed");
+  Serial.write(30002); // 30002 = Command, N/A, Execute Command: Disarm
```
Comment (Lastrellik) [inline src/Alarm.cpp]: Same as above issues

File: src/Alarm.cpp [hunk 5]
```diff
@@ -53,91 +80,138 @@ void Alarm::determineBasePhotoresistorReading(){
...
 }
+/**
+  Silences the alarm by setting _isSilenced to true
+*/
 void Alarm::silence(){
   _isSilenced = true;
-  Logger::Log("Alarm has been silenced");
+  Serial.write(30003); // 30003 = Command, N/A, Execute Command: Silence
```
Comment (Lastrellik) [inline src/Alarm.cpp]: Same as above issues

File: src/Buzzer.cpp [hunk 6]
```diff
@@ -1,45 +1,87 @@
...
+  Uses the int frequency to determine the pitch of the tone being passed to the
+    tone function.
+    This version of the soundTone Function does not use duration with the tone
+    function.
+*/
 void Buzzer::soundTone(int frequency) {
-  if(Properties::DEBUGGING_ACTIVE) Serial.println("Buzzer has started sounding");
+  //Need to take a look at while we're using properies for a Serail.println
+  //does not affected the tone
+  //if(Properties::DEBUGGING_ACTIVE) Serial.println("Buzzer has started sounding");
```
Comment (Lastrellik) [inline src/Buzzer.cpp]: Yeah we can remove this code. That should be fine.

File: src/Buzzer.cpp [hunk 7]
```diff
@@ -1,45 +1,87 @@
...
 void Buzzer::soundNegativeTone(){
   this->soundTone(400, 300);
 }
-
+/**
+  Stops the tone of the Buzzer sounding
+*/
 void Buzzer::stopTone() {
-  if(Properties::DEBUGGING_ACTIVE) Serial.println("Buzzer has stopped sounding");
+  //if(Properties::DEBUGGING_ACTIVE) Serial.println("Buzzer has stopped sounding");
```
Comment (Lastrellik) [inline src/Buzzer.cpp]: We can remove this line of code too.

File: src/Laser.cpp [hunk 8]
```diff
@@ -1,26 +1,45 @@
-/*
-  Laser.cpp - Simmple library for the laser module
+/**
+  Calico Home Security System, Laser.h
+  Purpose: Manges the lASER and creats a LASER object
```
Comment (Lastrellik) [inline src/Laser.cpp]: Few Typos here

File: src/Laser.cpp [hunk 9]
```diff
@@ -1,26 +1,45 @@
...
 void Laser::off(){
+  Serial.write(13352); // 13352 = Log, Debug, Laser has been turned off
   digitalWrite(Component::getPin(), LOW);
   _isOn = false;
 }
-
+//Need to think about removing toggle as we don't really use it
 void Laser::toggle(){
   if(_isOn){
     digitalWrite(Component::getPin(), LOW);
```
Comment (Lastrellik) [inline src/Laser.cpp]: We can remove it. I'm sure that's fine.

File: src/Laser.h [hunk 10]
```diff
@@ -1,5 +1,9 @@
-/*
-  Laser.h - Simple class for laser module
+/**
+  Calico Home Security System, Laser.h
+  Purpose: Manges the lASER and creats a LASER object
```
Comment (Lastrellik) [inline src/Laser.h]: Manges? Manages? Another Typo :P

**Review by Lastrellik — CHANGES_REQUESTED** [review]
Just a few changes here before we can merge it. Shouldn't take too long.

File: src/Laser.h (hunk 10 above)
Comment (Boomstick105) [reply src/Laser.h]: I'll get right on fixing these issues, I guess I was miss understanding the command function a little bit, shouldn't be to hard to update these real quick.

Comment (Boomstick105) [reply src/Laser.h]: Should be all updated now by getting rid of the 30000's and the small typos

File: docs/development/logging.md [hunk 11]
```diff
@@ -47,16 +47,53 @@ Message Translation Chart
...
-
+* [100 - 149] - Reserved for Alarm Class messages to be logged.
+* [150 - 199] - Reserved for Button messages to be logged.
+* [200 - 249] - Reserved for Buzzer messages to be logged.
+* [250 - 299] - Reserved for Component messages to be logged.
+* [300 - 349] - Reserved for ComponentTester messages to be logged.
+* [350 - 399] - Reserved for Laser messages to be logged.
+* [400 - 449] - Reserved for LED messages to be logged.
+* [450 - 499] - Reserved for Photoresistor messages to be logged.
+* [700 - 999] - Currently reserved for David Tille.
```
Comment (Lastrellik) [inline docs/development/logging.md:96]: I like this.

File: src/Component.h [hunk 12]
```diff
@@ -1,5 +1,11 @@
-/*
-  Component.h - Component superclass for component classes
+/**
+  Calico Home Security System, Component.h
+  Purpose: Handles creation of a Component object.
+    Which are used to determine the type of compoent Ex. "LED" and where it is
```
Comment (Lastrellik) [inline src/Component.h]: Typo in compoent

File: src/Laser.cpp [hunk 13]
```diff
@@ -1,32 +1,42 @@
...
 }
-
+/**
+  Base contructor
+*/
 Laser::Laser(){
 }
-
+/**
+  Turns the lASER on
```
Comment (Lastrellik) [inline src/Laser.cpp]: Should all letters in LASER be upper case?

**Review by Lastrellik — CHANGES_REQUESTED** [review]
So The changes all look good with the exception of a few typos. I also never loaded the code onto the arduino because it's currently plugged into the PI on the other side of the room. I'll load it up and look at it tonight.

**Comment by Lastrellik** [comment]
@Zeustopher can you load this code and make sure it functions? I think that removing the Laser::toggle() from the cpp but not from the header may cause it to break. I hate C++. My arduino is tied up with the PI at the moment.

**Comment by Boomstick105** [comment]
I forgot to remove that from the laser.h, I edit it a couple of times and then went back I can remove that from the resubmit real quick @Lastrellik

**Comment by Zeustopher** [comment]
@Boomstick105 - There are several build errors when I try and build this branch in PlatformIO. Is it building on properly on your machine?

**Comment by Boomstick105** [comment]
I'm getting errors from the properties and logger class because of a / and
an extra bracket. Just pushed the fix and it built fine.

**Comment by Lastrellik** [comment]
@Boomstick105 we should be able to remove the logger class. We're not building datapackets anymore, so there should be no problem with removing that one.

**Review by Zeustopher — APPROVED** [review]
I think there are some additional tweaks we can make to this as we move forward, but this Pull Request is starting to get large and already has quite a few commits to it. I vote that we merge it in as-is right now and work on those additional changes (like removing the Logger class) in another Pull Request.

If anyone else agrees go ahead and merge this pull request.

**Comment by Lastrellik** [comment]
Merged [#55](https://github.com/Lastrellik/Calico-Home-Security/pull/55)

**Pull request merged**